File: 백준/Silver/11651. 좌표 정렬하기 2/좌표 정렬하기 2.js
function solution(input) {
    let result = '';
    
    input.sort((a,b) => {
        if(a[1] === b[1]) {
            return a[0] - b[0];
        }
        return a[1] - b[1];
    }).forEach(el => {
      result += `${el[0]} ${el[1]}\n`;  
    })

    console.log(result);
}

const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,    
});

let input = [];
let list = [];

rl.on("line", function (line) {
   input.push(line)
}).on("close", function () {
    input.forEach((val) => {
       list.push(val.split(' ').map((el) => parseInt(el)));
    });
    list.shift();
    solution(list);
    process.exit();
});

File: 백준/Silver/11651. 좌표 정렬하기 2/좌표 정렬하기 2.py
"""
1. 입력 -> O(N): 100,000
2. FOR문 사용 시, 시간초과날 확률이 높음
3. 출력 -> 좌표 순서대로
"""

array = []
N = int(input())

for _ in range(N): # O(N)
    x, y = map(int, input().split())
    array.append((y, x))

array.sort(reverse=True)

for _ in range(N): # O(N)
    y, x = array.pop()
    print(x, y)